node/rpc/src/wallet_api.rs

```diff
@@ -0,0 +1,103 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use address::Address;
+use blockstore::BlockStore;
+use crypto::{Signature, SignatureType};
+use encoding::Cbor;
+use message::{SignedMessage, UnsignedMessage};
+use num_bigint::BigUint;
+use state_manager::StateManager;
+use thiserror::Error;
+use wallet::{KeyInfo, KeyStore, Wallet};
+
+#[derive(Debug, Error, PartialEq)]
+pub enum Error {
+    #[error("{0}")]
+    WalletError(String),
+}
```

Comment:
Is an error type necessary for this? What is the benefit?

Reply:
I used it to collect all the different errors that could be returned from the different methods.

Reply:
so there are probably better use cases for this:
1. You could have the error type be `Box<dyn std::error:Error>` so no conversion necessary
2. Could just have a `String` error type instead of this and just not wrap in this error enum